ubiconfig/_impl/loaders/local.py

```diff
@@ -40,16 +41,22 @@ def load(self, file_name, version=None):
 
                 If version is None, we should get it from its path.
         """
-        if version is None:
-            # get version form path, such as configs/ubi7.1/config.yaml, get
-            # ubi7.1
-            version = os.path.basename(os.path.dirname(file_name))
-
         if not self._isroot:
             file_path = os.path.join(self._path, file_name)
         else:
             file_path = file_name
 
+        if version is None:
+            # get version from path, such as configs/ubi7.1/config.yaml, get
+            # ubi7.1.
+            version = os.path.basename(os.path.dirname(os.path.abspath(file_path)))
+
+        if not re.search(r"ubi[0-9]{1}\.?[0-9]{0,2}$", version):
```

Comment:
hmm, 
are you sure of this? It makes only dot optional.
Shouldn't it be:
ubi[0-9]{1}(\\.[0-9]{0,2})?

Reply:
Hm, it should be r"ubi[0-9]{1}\.[0-9]{1,2}$|ubi[0-9]?$"

Reply:
what is allowed?
ubi7
ubi8
ubi7.1
ubi8.9
ubi7.10

I guess my regex would do the trick.

Reply:
@JayZ12138 
next time let's code formatter in comment, otherwise it can strip out some chars

I was wrong before:
the shortest correct should be:
`r'ubi[0-9](\.[0-9]{1,2})?$'`